### DISCUSSION TITLE
Combine rendered PDF with another PDF?

### DISCUSSION THREAD

**Discussion by singhpaul** [dompdf/dompdf#3175] [Q&A] [answered]
Is it possible to take a PDF object rendered with dompdf and append another PDF to the end? 

We thought this might be possible via the Canvas get_{{backend}} methods but not sure what this means: 

> Each implementation of Canvas provides it's own method for accessing the backend library. This allows you to call methods of the backend that are not represented in the Canvas API.
> 
>     get_cpdf (Cpdf)
>     get_pdflib (PDFLib)
>     get_image (GD)

Are all three backend libraries instantiated by default or is only one in use depending on the environment and/or object state? 

Ideally we could use the Canvas page_text() method on the combined PDF to add a watermark. 

Thank you.

**Comment by bsweeney** [comment, marked as answer]
Only a single backend is instantiated, and it's based on the [backend Option](https://github.com/dompdf/dompdf/blob/master/src/Options.php#L270-L281). The default is the Cpdf library.

None of the currently supported backend libraries have the ability to append an existing PDF to an existing document. You'd want to utilize another library, such as [FPDF](https://github.com/Setasign/FPDF) to combine documents.

**Comment by singhpaul** [comment]
Thank you, much appreciated.
